app/lib/three_scale/search/cinstances_searcher.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module ThreeScale::Search::CinstancesSearcher
+  def find_service(id = params[:service_id])
```

Comment:
Does it belong to `ThreeScale::Search::CinstancesSearcher` module?

Reply:
Yes, because it searches from a service:
```
def searcher(search = ThreeScale::Search.new(params[:search] || params))
    # ...
    search.service_id = @service.id if find_service
    # ...
  end
```
And it uses the content of this method to find the service only if the service is not already set somewhere 😄,  for the `@service ||=`

Reply:
Speaking with Hery right now on Slack about how to solve this. I will write conclusions here once I have them 😄